nav2_toolkit/include/nav2_toolkit/pose_saver_node.hpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2025 Pranav Kolekar
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_TOOLKIT__POSE_SAVER_NODE_HPP_
+#define NAV2_TOOLKIT__POSE_SAVER_NODE_HPP_
+
+#include <yaml-cpp/yaml.h>
+#include <memory>
+#include <string>
+#include <rclcpp/rclcpp.hpp>
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <nav2_msgs/srv/set_initial_pose.hpp>
+#include <std_srvs/srv/trigger.hpp>
+#include "nav2_ros_common/service_client.hpp"
+
+
+namespace nav2_toolkit
+{
+
+/**
+ * @class PoseSaverNode
+ * @brief A utility node to periodically save and restore the robot's pose from a YAML file.
+ *
+ * This node subscribes to `amcl_pose`, stores the latest pose to a file periodically,
+ * and can also restore that pose by calling the `set_initial_pose` service.
+ *
+ * Services:
+ * - `start_pose_saver`: Start periodic pose saving.
+ * - `stop_pose_saver`: Stop periodic pose saving.
+ * - `localise_at_last_known_position`: Restore the saved pose via service call.
+ */
+class PoseSaverNode : public rclcpp::Node
```

Comment:
The methods here need full doxygen on the `brief` paramater arguments and returns (generally speaking, look at some Nav2 headers and copy the style / formatting, consistency is really important in projects of this size so that developers can open any file and feel like its consistent and understood)